cachi2/core/package_managers/npm.py

```diff
@@ -476,6 +502,104 @@ def _get_npm_dependencies(
+        return "git"
+    else:
+        # Anything else should be considered as registry version or range of registry versions or latest
+        return "registry"
+
+
+def update_package_lock_with_local_paths(
+    download_paths: Dict[NormalizedUrl, RootedPath],
+    package_lock_file: ProjectFile,
+    output_dir: RootedPath,
+) -> ProjectFile:
+    """Replace packages resolved URLs with local paths.
+
+    Update package-lock.json file in a way it can be used in isolated environment (container)
+    without internet connection. All package/dependency URLs will be replaced for
+    local paths to downloaded dependencies.
+
+    :param download_paths:
+    :param package_lock_file: File path to package-lock.json
+    :param output_dir: Output directory with fetched dependencies
+    """
+    package_lock = PackageLock.from_file(RootedPath(package_lock_file.abspath))
+
+    if package_lock.lockfile_version > 1:
+        for package in package_lock._get_packages():
+            # Remove integrity for git sources, their integrity checksum will change when
+            # constructing tar archive from cloned repository
+            url = _normalize_resolved_url(package.resolved_url)
+            if _classify_resolved_url(url) == "git":
+                if package.integrity:
+                    package.integrity = ""
+
+            # Update all type of dependencies except file:
+            if _classify_resolved_url(url) != "file":
+                templated_abspath = str(download_paths[url].path).replace(
+                    str(output_dir.path), "${output_dir}"
+                )
+                package.resolved_url = f"file://{templated_abspath}"
+
+            if package.dependencies:
+                for dependency, dependency_version in package.dependencies.items():
+                    if get_type_of_dependency(dependency_version) in ["https", "git"]:
+                        package.dependencies.update({dependency: ""})
+
+        # TODO: Tried to use get_main_package method but it failed horribly with error:
+        #   target is outside `package-lock.json` Error: PathOutsideRoot: Joining path '..' to `package-lock.json` target is outside
+        main_package = package_lock._lockfile_data["packages"][""]
+        if main_package["dependencies"]:
+            for dependency, dependency_version in main_package["dependencies"].items():
+                if get_type_of_dependency(dependency_version) in ["https", "git"]:
+                    main_package["dependencies"].update({dependency: ""})
+
+    return ProjectFile(abspath=package_lock_file.abspath, template=str(package_lock._lockfile_data))
+
+
+def update_package_json_files(workspaces: list[str], source_dir: RootedPath) -> list[ProjectFile]:
+    """Set dependencies to empty string in package.json files.
+
+    :param workspaces: list of workspaces paths
+    :param source_dir: Source directory with repository content
```

Comment:
Is it the root of the repository? It should be the the package (sub)directory, no?